### PR TITLE
Cash App Pay component

### PR DESCRIPTION
**Pull request by jreij**
## Description
Adding a new component to support Cash App Pay in both the normal and stored flows.

## Checklist 
- [x] Changes are tested manually
- [x] Add relevant labels to PR  

COAND-634

### CODE REVIEW

File: cashapppay/src/main/java/com/adyen/checkout/cashapppay/DefaultCashAppPayDelegate.kt [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+            getOnFileAction(cashAppParams, outputData),
+        )
+
+        if (actions.isEmpty()) {
+            throw ComponentException("Cannot launch Cash App Pay, you need to either pass an amount or store the shopper account")
+        }
+
+        withContext(Dispatchers.IO) {
+            // must be called from a background thread according to Cash App Pay docs
+            cashAppPay.createCustomerRequest(actions, cashAppParams.returnUrl)
```
Comment (OscarSpruit) [inline cashapppay/src/main/java/com/adyen/checkout/cashapppay/DefaultCashAppPayDelegate.kt:85]: Does this actually do IO work? If not, the default dispatcher would be better

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt [hunk 2]
```diff
@@ -253,16 +261,21 @@ internal fun getComponentFor(
     dropInConfiguration: DropInConfiguration,
     amount: Amount
 ): PaymentComponent<PaymentComponentState<in PaymentMethodDetails>, Configuration> {
-
+    val context = fragment.requireContext()
```
Comment (OscarSpruit) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:264]: Is it possible that we call this before the fragment is actually created or after it's been destroyed?

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt [hunk 3]
```diff
@@ -285,87 +298,98 @@ internal fun getComponentFor(
...
```
Comment (OscarSpruit) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:301]: Same question here

File: drop-in/src/main/res/layout/fragment_cash_app_pay_component.xml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    <TextView
+        android:id="@+id/header"
+        style="@style/AdyenCheckout.TextAppearance.HeaderTitle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textIsSelectable="false"
+        tools:text="Header" />
+
+    <LinearLayout
```
Comment (OscarSpruit) [inline drop-in/src/main/res/layout/fragment_cash_app_pay_component.xml:29]: This `LinearLayout` is unnecessary I think

File: cashapppay/src/main/java/com/adyen/checkout/cashapppay/DefaultCashAppPayDelegate.kt (hunk 1 above)
Comment (jreij) [reply cashapppay/src/main/java/com/adyen/checkout/cashapppay/DefaultCashAppPayDelegate.kt:85]: This is from the Cash App SDK, I think we should explicitly specify `Dispatchers.IO` right?
```kotlin
  /**
   * Create customer request given list of [CashAppPayPaymentAction].
   *
   * Must be called from a background thread.
   * ...
   */
  @WorkerThread
  fun createCustomerRequest(paymentActions: List<CashAppPayPaymentAction>, redirectUri: String?)
```

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt (hunk 2 above)
Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:264]: Seeing where we call it from now, it's mostly in fragments inside `onCreate`

File: cashapppay/src/main/java/com/adyen/checkout/cashapppay/DefaultCashAppPayDelegate.kt (hunk 1 above)
Comment (OscarSpruit) [reply cashapppay/src/main/java/com/adyen/checkout/cashapppay/DefaultCashAppPayDelegate.kt:85]: Checked the code and it does a network call, so `IO` is perfect 👍

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**